documentation/architecture.md

```diff
@@ -0,0 +1,139 @@
+| `failures_before_alerting` | `int` | Number of consecutive failures before an alert is triggered. |
+| `alert_slack_channels` | `list<str>` | List of Slack channels to send alerts to. |
+| `alert_emails` |`list<str>` | List of email addresses to send alerts to. |
+| `realert_interval_mins` | `int` | Minimum time in minutes between consecutive alerts for the same issue. |
+
+#### **Here is an example of what it could look like:**
+
+| project_id | slug | state | url | frequency_mins | next_due_at | timeout_secs | assertions | failures_before_alerting | alert_slack_channels | alert_emails | realert_interval_mins |
+|------------|------|-------|-----|----------------|-------------|--------------|------------|-------------------------|---------------------|--------------|----------------------|
+| `a1k9...` | `homepage-check` | `up` | `https://example.com/home-page` | 5 | `1730851200` | 30 | `{"response.code": 200, "response.time": {"$lt": 2000}}` | 3 | `["#alerts", "#ops"]` | `["ops@example.com"]` | 60 |
+| `a1k9...` | `api-health` | `down` | `https://api.example.com/health` | 1 | `1730851260` | 10 | `{"response.code": 200}` | 1 | `["#critical"]` | `["oncall@example.com"]` | 30 |
+| `b45d...` | `login-endpoint` | `paused` | `https://app.example.com/login` | 10 | `1730851800` | 15 | `{"response.code": {"$in": [200, 302]}}` | 2 | `[]` | `[]` | 120 |
+
+- Assertions here may seems confusing at first! But if we look carefully, you can see there is a map from the name of the assertion to what the assertion is. In the first example it checks if the response code is a 200, and it also chekcs if the response time is `$lt`(less than) 2000 ms.
+
+## UptimeLog
+The `UptimeLog` model stores the results of each individual uptime check.
+
+| Field | Type | Description |
+|---|---|---|
+| `monitor_id` | `uuid` | Partition key, links to the `UptimeMonitor` this log belongs to. |
+| `timestamp` | `int` | Sort key, epoch timestamp of when the check was performed. |
+| `status` | `enum` | Result of the check (`up` or `down`). |
+| `resp_code` | `int` | HTTP response code received. |
+| `latency_secs` | `float` | Latency of the HTTP request in seconds (can represent milliseconds, as DynamoDB numbers are floats). |
+
+**Here is an example of what it could look like:**
+
+| monitor_id | timestamp | status | resp_code | latency_secs |
+|------------|-----------|--------|-----------|--------------|
+| `a1k9...#homepage-check` | `1730851200` | `up` | 200 | 0.450 |
+| `a1k9...#homepage-check` | `1730850900` | `up` | 200 | 0.420 |
+| `a1k9...#homepage-check` | `1730850600` | `down` | 500 | 1.200 |
+| `a1k9...#api-health` | `1730851260` | `down` | 500 | 0.800 |
+| `a1k9...#api-health` | `1730851200` | `down` | 503 | 1.050 |
+| `b45d...#login-endpoint` | `1730851800` | `up` | 302 | 0.320 |
```

Comment:
The monitor uuid and url should not be combined here in the `monitor_id` field. Just the uuid.